plutus-playground-client/src/Halogen/Chartist.purs

```diff
@@ -0,0 +1,95 @@
+module Halogen.Chartist
+       ( chartist
+       , ChartistEffects
+       , ChartistQuery(..)
+       , ChartistMessage(..)
+       ) where
+
+import Chartist (CHARTIST, Chart, ChartistData, ChartistOptions, updateData)
+import Chartist as Chartist
+import Control.Applicative (pure)
+import Control.Bind (bind, discard, (>>=))
+import Control.Monad.Aff.Class (class MonadAff)
+import Control.Monad.Eff.Class (liftEff)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import DOM (DOM)
+import Data.Function (($))
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.NaturalTransformation (type (~>))
+import Data.Unit (unit)
+import Halogen (RefLabel(..))
+import Halogen as H
+import Halogen.HTML (ClassName(..))
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties (classes)
+import Halogen.HTML.Properties as HP
+
+type ChartistState =
+  { chart :: Maybe Chart
+  }
+
+data ChartistQuery a
+  = Init ChartistOptions a
+  | SetData ChartistData a
+
+data ChartistMessage
+  = Initialized
+
+type HTML = H.ComponentHTML ChartistQuery
+
+type DSL g = H.ComponentDSL ChartistState ChartistQuery ChartistMessage g
+
+type ChartistEffects eff =
+  ( chartist :: CHARTIST
+  , dom :: DOM
+  , exception :: EXCEPTION
+  | eff
+  )
+
+chartist
+  :: forall eff g.
+  MonadAff (ChartistEffects eff) g
+  => ChartistOptions
+  -> H.Component HH.HTML ChartistQuery ChartistData ChartistMessage g
+chartist options = H.lifecycleComponent
+  { initialState: \_ -> { chart: Nothing }
+  , render
+  , eval
+  , initializer: Just $ H.action $ Init options
+  , finalizer: Nothing
+  , receiver: HE.input SetData
+  }
+
+eval
+  ∷ ∀ eff g
+  . (MonadAff (ChartistEffects eff) g)
+  ⇒ ChartistQuery ~> (DSL g)
```

Comment:
OMG you're going with the unicode!!! 😮

Reply:
Noooo! Sorry, this is a copy & paste from the halogen-echarts wrapper. Will fix. :-)